src/appleseed/renderer/modeling/project/projectfilewriter.cpp

```diff
@@ -113,8 +114,8 @@ namespace renderer
 namespace
 {
     // Floating-point formatting settings.
-    const char* MatrixFormat     = "%.15f";
-    const char* ColorValueFormat = "%.6f";
+    std::string MatrixFormat     = "%." + std::to_string(std::numeric_limits<double>::max_digits10) + "f";
```

Comment:
I appreciate the effort here but I'd be happy to keep the simpler, lighter-weight solution we had before, just with updated precision. The values of `std::numeric_limits<double>::max_digits10()` and `std::numeric_limits<float>::max_digits10()` are fixed and equal on all platforms (right?).

Reply:
As long as the IEEE 754 standard is used (which is recommended by the C++ standard, but not required!) it is. If you want to keep it simple one could argue, there are just too few platforms (which might also not be able to run appleseed at all) which differ here.
I thought, that way we are safe, if things change somewhen in the future (not predictable yet), or we got that one person trying to render on his washing mashine. :D

Reply:
Moreover, `std::numeric_limits<T>::max_digits10` isn't a function, but a `static constexpr` and thus should benefit a lot from optimization.
I will run some tests tomorrow to check how strong the performance is affected by using these calls in comparison to a fixed expression.

Reply:
Performance won't be affected. It's just that there's little value in using that more complex expression when `%.17f` is shorter, clearer and sufficient :) But the thoroughness is definitely appreciated!

Reply:
Allright, changed to the more readable version.